src/components/FlatParameters/FlatParameters.js

```diff
@@ -0,0 +1,265 @@
+                            onChange={(e, value) => onFieldChange(value, param)}
+                            value={value}
+                            renderTags={(value, getTagProps) =>
+                                value.map((option, index) => (
+                                    <Chip
+                                        variant="outlined"
+                                        label={option}
+                                        {...getTagProps({ index })}
+                                    />
+                                ))
+                            }
+                            renderInput={(options) => (
+                                <TextField {...options} variant="standard" />
+                            )}
+                        />
+                    );
+                }
+            // else fallthrough to default
+            case 'STRING':
+                if (param.possibleValues) {
+                    return (
+                        <Autocomplete
+                            fullWidth
+                            disableClearable
+                            options={param.possibleValues}
+                            onChange={(e, value) => onFieldChange(value, param)}
+                            value={value}
+                            renderInput={(options) => (
+                                <TextField {...options} variant="standard" />
+                            )}
+                        />
+                    );
+                }
+            // else fallthrough to default
+            default:
+                return (
+                    <TextField
+                        fullWidth
+                        defaultValue={value}
+                        onFocus={() => onUncommited(param, true)}
+                        onBlur={() => onUncommited(param, false)}
+                        onChange={(e) => onFieldChange(e.target.value, param)}
+                        variant={'standard'}
+                    />
+                );
+        }
+    };
+
+    const renderAsList = () => {
+        return (
+            <List className={classes.paramList}>
+                {paramsAsArray.map((param) => (
+                    <Tooltip
+                        title={param.description}
+                        enterDelay={1200}
+                        key={param.name}
+                    >
+                        <ListItem
+                            key={param.name}
+                            className={classes.paramListItem}
+                        >
+                            <Typography className={classes.paramName}>
+                                {param.name.slice(prefix.length)}
+                            </Typography>
+                            {renderField(param)}
+                        </ListItem>
+                    </Tooltip>
+                ))}
+            </List>
+        );
+    };
+
+    return renderAsList();
```

Comment:
why not returning directly the JSX then ?

Reply:
Because I had another way to render (render as grid) and I had the possibility to toggle easily this way.